lib/CL/devices/cuda/pocl-cuda.c

```diff
@@ -559,30 +561,33 @@ pocl_cuda_alloc_mem_obj (cl_device_id device, cl_mem mem, void *host_ptr)
 
   if (flags & CL_MEM_USE_HOST_PTR)
     {
-#if defined __arm__
+      if (!((pocl_cuda_device_data_t *)device->data)->supports_cu_mem_host_register)
+        {
           /* cuMemHostRegister is not supported on ARM.
```

Comment:
This sounds it could even be a build-time property if it never works on arm hosts?

Reply:
```suggestion
          /* cuMemHostRegister is not supported on some ARM devices like the Nano, but supported on Xavier.
```